src/ape/api/transactions.py

```diff
@@ -46,6 +47,31 @@ class TransactionAPI(BaseInterfaceModel):
     class Config:
         allow_population_by_field_name = True
 
+    @validator("gas_limit", pre=True)
+    def validate_gas_limit(cls, value):
+        if value is None:
+            if not cls.network_manager.active_provider:
+                raise NetworkError("Must be connected to use default gas config.")
+
+            value = cls.network_manager.active_provider.network.gas_limit
```

Comment:
OK - I figured out why things are not working on this PR. Thins are happening in the wrong order.

The default network config for `gas_limit` is **not** honored here (only in this PR).
The gas estimation logic in `prepare_transaction()` triggers _before_ this validation logic here in the txn model.
Right now, it will always use `auto-estimate`, regardless of what is specified in the config file, and that is why there are failing tests.

Reply:
hmm, maybe it is actually that there are conflicting validators in `ape_ethereum`. So maybe not a timing issue but a conflict.